dix/devices.c

```diff
@@ -1954,19 +1946,25 @@ ProcGetKeyboardMapping(ClientPtr client)
         .length = count
     };
 
+    x_rpcbuf_t rpcbuf = { .swapped = client->swapped, .err_clear = TRUE };
+    x_rpcbuf_write_CARD32s(
+        &rpcbuf,
+        &syms->map[syms->mapWidth * (stuff->firstKeyCode - syms->minKeyCode)],
+        count);
```

Comment:
It was `size = count * sizeof(KeySym)` Is `sizeof(KeySym)` always 4 bytes? Or it's 8 bytes?

Reply:
Later they called `SwapLongs` so it suggests that it's indeed 32bit.

But if `KeySym` defined like this in X11/X.h:
```
typedef unsigned long int XID;
typedef XID KeySym;
```
then it can be 64bit?

Reply:
ahh, you're right, it's conditional and it's always 32bit

I just quickly googled [this](https://refspecs.linuxbase.org/LSB_3.1.1/LSB-Desktop-generic/LSB-Desktop-generic/libx11-ddefs.html) but it's probably too old, that's why I was confused.

The real headers check `_XSERVER64`.

Reply:
yeah, just checked it. in case we're inside a 64bit Xserver, it's defining it to CARD32.